### PR TITLE
move "org.camaraproject.geofencing.v0.subscription-ends" out of /subscriptions - request

### LINKED ISSUE #113: Geofencing API - Subscription-ends should not a subscribe-able event type
**Problem description**
The `org.camaraproject.geofencing.v0.subscription-ends` notification type did not require a subscription. Indeed this notification type should be always send when the subscription terminates. As such this type must not be listed in the POST /Subscriptions (but this type could be of course used in the notification)

**Expected behavior**
In the yaml have a separate array for the event type in the subscription and the one for the notification in order to not allow `org.camaraproject.geofencing.v0.subscription-ends`  in the `POST /souscriptions`

**Alternative solution**

**Additional context**

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:
* correction

#### What this PR does / why we need it:

- Removes the event "org.camaraproject.geofencing.v0.subscription-ends" out of the _/subscriptions_ - request. 
Only use this event for CloudEvent - callbacks.
- Replace single-quotes with double-quotes

#### Which issue(s) this PR fixes:

Fixes #113

### CODE REVIEW

**Review by bigludo7 — APPROVED** [review]
LGTM 
Thanks @maxl2287

File: code/API_definitions/geofencing.yaml [hunk 1]
```diff
@@ -441,26 +441,33 @@ components:
...
         area-left - Event triggered when the device leaves the given area
-        subscription-ends - Event triggered when the subscription ends
-      type: string
       enum:
         - org.camaraproject.geofencing.v0.area-entered
         - org.camaraproject.geofencing.v0.area-left
-        - org.camaraproject.geofencing.v0.subscription-ends
+
+    SubscriptionEventType:
+      allOf:
```
Comment (jlurien) [inline code/API_definitions/geofencing.yaml:464]: We cannot "extend" an enum this way, as allOf requires compliance to both schemas, and any of the specific values cannot comply with both schemas at the same time. We have to define a new enum with the 3 values, same model as in https://github.com/camaraproject/DeviceStatus/blob/main/code/API_definitions/device-status.yaml#L568

**Review by jlurien — CHANGES_REQUESTED** [review]

File: code/API_definitions/geofencing.yaml [hunk 2]
```diff
@@ -441,26 +441,38 @@ components:
...
+        area-entered - Event triggered when the device enters the given area
+        
+        area-left - Event triggered when the device leaves the given area
+        
+        subscription-ends - Event triggered when the subscription ends
       enum:
         - org.camaraproject.geofencing.v0.area-entered
         - org.camaraproject.geofencing.v0.area-left
-        - org.camaraproject.geofencing.v0.subscription-ends
+        - org.camaraproject.device-status.v0.subscription-ends
```
Comment (jlurien) [inline code/API_definitions/geofencing.yaml:475]:
```suggestion
        - org.camaraproject.geofencing.v0.subscription-ends
```

**Review by jlurien — CHANGES_REQUESTED** [review]

**Review by jlurien — APPROVED** [review]
LGTM

**Review by bigludo7 — APPROVED** [review]
LGTM

**Pull request merged**